src/System.Private.ServiceModel/tests/Scenarios/Encoding/Encoders/TextEncodingTests.cs

```diff
@@ -48,4 +48,36 @@ public static void SameBinding_SecurityModeNone_Text_EchoString_Roundtrip()
 
         Assert.True(errorBuilder.Length == 0, "Test case FAILED with errors: " + errorBuilder.ToString());
     }
+
+    [Fact]
+    [OuterLoop]
+    public static void TextMessageEncoder_WrongContentTypeResponse_Throws_ProtocolException()
+    {
```

Comment:
It would be good to add a positive test to demonstrate what happens with a known good content type. And possibly another test where the content type is parsable but not expected.   In fact, what does happen if the WCF service sets a content type that does not agree with what the client expected (e.g. saying it returned json when it returned xml)?